proposals/NNNN-unlock-existential-types-for-all-protocols.md

```diff
@@ -0,0 +1,405 @@
+# Unlock Existentials with Fixed Associated Types
+
+* Proposal: [SE-NNNN](NNNN-unlock-existential-types-for-all-protocols.md)
+* Authors: [Anthony Latsis](https://github.com/AnthonyLatsis), [Filip Sakel](https://github.com/filip-sakel), [Suyash Srijan](https://github.com/theblixguy)
+* Review Manager: TBD
+* Status: **Awaiting Review**
+* Implementation: [apple/swift#33767](https://github.com/apple/swift/pull/33767)
+
+## Introduction
+
+Swift currently offers the ability for protocols that meet certain criteria to be used as types. Trying to use an unsupported protocol as a type yields the error: [the protocol] can only be used as a generic constraint because it has 'Self' or associated type requirements. This proposal aims to relax this artificial constraint imposed on such protocols.
+
+## Motivation
+
+Existential types are types synthesized by the compiler that can wrap any value of a type which conforms to a given _protocol_, _class_, or _composition of the above_. They are an incredibly useful feature in Swift for specific use cases that involve dynamicity. Thus, by being able to wrap different types that conform to a specified protocol, users and, more often, library authors can alleviate a lot of boilerplate code. That is, in certain use cases creating a custom existential type can be quite a daunting task due to its complexity; not to mention, that it introduces a lot of difficult-to-understand and hard-to-maintain code.
```

Comment:
> I think omitting some of the sentence connectors makes for a more streamline reading experience.

Okay, I'll remove them.

> Did you mean type erasure?

Yeah, but I think the notion of compiler-generated and user-written existentials is easy to grasp.; not to mention that, at least in my opinion, it draws a simple distinction between the two. However, if you think that it will confuse the reader then I don't mind changing it to "type erasure" instead.

> Another con would be additional API/ABI surface and ensuing resilience limitations.

Great point! I'll add it to the proposal.